File: web3rpc/rpc-specs/code-samples/javascript/eth/block/getBlockByNumber.js
const { JsonRpcProvider } = require("@kaiachain/ethers-ext");
(() => {
  const provider = new JsonRpcProvider("https://public-en-kairos.node.kaia.io");

  const blockNumber = 1;
  const transactionObject = true;
  provider.eth
    .getBlockByNumber(
      blockNumber,
      transactionObject,
      {},
      (err, data, response) => {},
    )
    .then((data) => {
      console.log(data);
    });
})();

File: web3rpc/rpc-specs/code-samples/python/eth/block/getBlockByNumber.py
from web3 import Web3
from web3py_ext import extend

host = "https://public-en-kairos.node.kaia.io"

blockTag = "0xd0054e"
transactionObject = False

w3 = Web3(Web3.HTTPProvider(host))
eth_response = w3.eth.get_block_by_number(blockTag, transactionObject)

print(eth_response)
